File: java codes/2962- Count Subarrays Where Max Element Appears at Least K Times.java
class Solution {
    public long countSubarrays(int[] nums, int k) {
        int n = nums.length;
        int maxE = Arrays.stream(nums).max().getAsInt();

        long ans = 0;
        int count = 0;
        int i = 0, j = 0;

        while (j < n) {
            if (nums[j] == maxE) {
                count++;
            }
            while (i <= j && count >= k) {
                ans += (n - j);

                if (nums[i] == maxE) {
                    count--;
                }
                i++;
            }
            j++;
        }

        return ans;
    }
}

File: python codes/2962- Count Subarrays Where Max Element Appears at Least K Times.py
class Solution(object):
    def countSubarrays(self, nums, k):
        n = len(nums)
        maxE = max(nums)

        ans = 0
        count = 0
        i = 0
        j = 0

        while j < n:
            if nums[j] == maxE:
                count += 1
            while i <= j and count >= k:
                ans += (n - j)
                if nums[i] == maxE:
                    count -= 1
                i += 1
            j += 1

        return ans
